controllers/restore_controller.go

```diff
@@ -60,92 +63,108 @@ func (r *RestoreReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	restoreLogger := log.FromContext(ctx)
 	restore := &v1beta1.Restore{}
 
-	restoreLogger.Info(fmt.Sprintf(">> Enter reconcile for Restore CR name=%s (namespace: %s)", req.NamespacedName.Name, req.NamespacedName.Namespace))
+	//restoreLogger.Info(fmt.Sprintf(">> Enter reconcile for Restore CR name=%s (namespace: %s)", req.NamespacedName.Name, req.NamespacedName.Namespace))
 
 	if err := r.Get(ctx, req.NamespacedName, restore); err != nil {
-
-		// check if this is a NotFound error
-		if !k8serr.IsNotFound(err) {
-			restoreLogger.Error(err, "unable to fetch Restore CR")
-		}
-
-		restoreLogger.Info("Restore CR was not created in the %s namespace", req.NamespacedName.Namespace)
-		return ctrl.Result{RequeueAfter: restoreRequeueInterval}, client.IgnoreNotFound(err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var (
-		err           error
-		veleroRestore *veleroapi.Restore
-	)
-	veleroRestore, err = r.submitAcmRestoreSettings(ctx, restore, r.Client)
-	if veleroRestore != nil {
-		restore.Status.VeleroRestore = veleroRestore
-	}
-	if err != nil {
-		msg := fmt.Errorf("unable to create Velero restore for %s: %v", restore.Name, err)
-		restoreLogger.Error(err, err.Error())
-		restore.Status.LastMessage = msg.Error()
-		restore.Status.Phase = "Failed"
+	// retrieve the velero restore (if any)
+	veleroRestoreList := veleroapi.RestoreList{}
+	if err := r.List(ctx, &veleroRestoreList, client.InNamespace(req.Namespace), client.MatchingFields{restoreOwnerKey: req.Name}); err != nil {
+		restoreLogger.Error(err, "unable to list velero restores for restore %s/%s", req.Namespace, req.Name)
+		return ctrl.Result{}, err
 	}
-	restoreLogger.Info(fmt.Sprintf("<< EXIT reconcile for Restore resource name=%s (namespace: %s)", req.NamespacedName.Name, req.NamespacedName.Namespace))
 
-	if isRestoreFinsihed(restore) {
-		return ctrl.Result{}, errors.Wrap(r.Client.Status().Update(ctx, restore), "could not update status")
+	switch {
+	case len(veleroRestoreList.Items) == 0:
+		veleroRestore, err := r.initVeleroRestore(ctx, restore)
+		if err != nil {
+			restoreLogger.Error(err, "unable to create velero restore for restore %s/%s", req.Namespace, req.Name)
+			return ctrl.Result{}, err
+		}
+		if err = r.Create(ctx, veleroRestore, &client.CreateOptions{}); err != nil {
+			restoreLogger.Error(err, "unable to create velero restore for restore %s/%s", req.Namespace, req.Name)
+
+			return ctrl.Result{}, err
+		}
+		r.Recorder.Event(restore, v1.EventTypeNormal, "Velero Restore created:", fmt.Sprintf("%s/%s", veleroRestore.Namespace, veleroRestore.Name))
+		restore.Status.VeleroRestore = veleroRestore.DeepCopy()
+
+	case len(veleroRestoreList.Items) == 1:
+		if isRestoreFinsihed(restore) {
+			r.managedClustersHandler(restore)
```

Comment:
I'm a little confused about this call, as the method defined below just returns a (true, err)

Reply:
Sure, I'm writing this function in another PR: at the moment. Goal of this WIP was to have only a review of REST APIs.